spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -172,171 +149,69 @@ public RedisLockRegistry(RedisConnectionFactory connectionFactory, String regist
-	 * Used for obtaining same lock object within same thread and key.
-	 * To avoid memory leaks lock objects without actual lock are kept as weak references.
-	 * After gaining the actual lock, lock object moves from weak reference to hard reference and vise a versa.
-	 */
-	private Collection<RedisLock> getWeakThreadLocks() {
-		Set<RedisLock> locks = this.weakThreadLocks.get();
-		if (locks == null) {
-			locks = Collections.newSetFromMap(new WeakHashMap<RedisLock, Boolean>());
-			this.weakThreadLocks.set(locks);
-		}
-		return locks;
-	}
-
-	/**
-	 * Hard referenced locks, lock is kept here when actual lock is gained.
-	 */
-	private Collection<RedisLock> getHardThreadLocks() {
-		List<RedisLock> locks = this.hardThreadLocks.get();
-		if (locks == null) {
-			locks = new LinkedList<RedisLock>();
-			this.hardThreadLocks.set(locks);
-		}
-		return locks;
-	}
-
-	private RedisLock findLock(Collection<RedisLock> locks, Object key) {
-		if (locks != null) {
-			for (RedisLock lock : locks) {
-				if (lock.getLockKey().equals(key)) {
-					return lock;
-				}
-			}
-		}
-		return null;
-	}
-
-	private void toHardThreadStorage(RedisLock lock) {
-		if (this.weakThreadLocks.get() != null) {
-			this.weakThreadLocks.get().remove(lock);
-		}
-
-		getHardThreadLocks().add(lock);
-
-		//clean up
-		if (this.weakThreadLocks.get() != null && this.weakThreadLocks.get().isEmpty()) {
-			this.weakThreadLocks.remove();
-		}
-	}
-
-	private void toWeakThreadStorage(RedisLock lock) {
-		//to avoid collection creation on existence check use direct fields
-		if (this.hardThreadLocks.get() != null) {
-			getHardThreadLocks().remove(lock);
-		}
-
-		if (this.useWeakReferences) {
-			getWeakThreadLocks().add(lock);
-		}
-
-		//clean up
-		if (this.hardThreadLocks.get() != null && this.hardThreadLocks.get().isEmpty()) {
-			this.hardThreadLocks.remove();
-		}
 	}
 
 	@Override
 	public Lock obtain(Object lockKey) {
 		Assert.isInstanceOf(String.class, lockKey);
-
-		//try to find the lock within hard references
-		RedisLock lock = findLock(this.hardThreadLocks.get(), lockKey);
-
-		/*
-		 * If the lock is locked, check that it matches what's in the store.
-		 * If it doesn't, the lock must have expired.
-		 */
-		if (lock != null && lock.thread != null) {
-			RedisLock lockInStore = this.redisTemplate.boundValueOps(this.registryKey + ":" + lockKey).get();
-			if (lockInStore == null || !lock.equals(lockInStore)) {
-				try {
-					lock.unlock();
-				}
-				catch (Exception e) {
-					if (logger.isWarnEnabled()) {
-						logger.warn("Lock was released due to expiration. A new one will be obtained...", e);
-					}
-				}
-				if (this.hardThreadLocks.get() != null) {
-					this.hardThreadLocks.get().remove(lock);
-				}
-				if (this.weakThreadLocks.get() != null) {
-					this.weakThreadLocks.get().remove(lock);
-				}
-				lock = null;
+		String path = (String) lockKey;
+		RedisLock lock;
+		synchronized (this.locks) {
```

Comment:
👍 without comments.
We also can consider to address the same in the `JdbcLockRegistry` and `ZookeeperLockRegistry`